Userland/Libraries/LibWeb/WebAudio/AnalyserNode.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2024, Noah Bright <noah.bright.1@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibJS/Forward.h>
+#include <LibWeb/WebAudio/AudioNode.h>
+
+namespace Web::WebAudio {
+
+// https://webaudio.github.io/web-audio-api/#AnalyserOptions
+struct AnalyserOptions : AudioNodeOptions {
+    unsigned long fft_size { 2048 };
+    double max_decibels { -30 };
+    double min_decibels { -100 };
+    double smoothing_time_constant { 0.8 };
+};
+
+// https://webaudio.github.io/web-audio-api/#AnalyserNode
+class AnalyserNode final : public AudioNode {
+    WEB_PLATFORM_OBJECT(AnalyserNode, AudioNode);
+    JS_DECLARE_ALLOCATOR(AnalyserNode);
+
+public:
+    virtual ~AnalyserNode() override;
+
+    WebIDL::ExceptionOr<void> get_float_frequency_data(JS::Float32Array& array);
+    WebIDL::ExceptionOr<void> get_byte_frequency_data(JS::Uint8Array& array);
+    WebIDL::ExceptionOr<void> get_float_time_domain_data(JS::Float32Array& array);
+    WebIDL::ExceptionOr<void> get_byte_time_domain_data(JS::Uint8Array& array);
```

Comment:
offending definitions

Reply:
Have a look at the generated code for these IDL definitions:

```
undefined getFloatFrequencyData (Float32Array array);
```

Going by your error message, it seems that the IDL generator hands you a `Handle<WebIDL::BufferSource>`. So your methods must take that as a parameter, and then check the type of the array.

Reply:
This old PR from domenic suggests using "write" to extract the bytes from the BufferSource into something the implementation can use. https://github.com/WebAudio/web-audio-api/pull/2358/files

It might be worth opening another issue on the WebAudio specification to see what the best resolution for this is, as it looks like the issue referenced in the spec for https://webaudio.github.io/web-audio-api/#dom-analysernode-getfloatfrequencydata is very old.